File: flutter/lib/models/task.dart
class Task {
  String id;
  String title;
  String description;
  String taskHour;
  String taskDay;
  String? userId;

  Task({
    required this.id,
    required this.title,
    required this.description,
    required this.taskHour,
    required this.taskDay,
  });

  factory Task.fromJson(Map<String, dynamic> json) {
    return Task(
      id: json['id'],
      title: json['title'],
      description: json['description'],
      taskHour: json['taskHour'].substring(0, 5),
      taskDay: json['taskDay']
    );
  }
}

class TaskDTO {
  String title;
  String description;
  String taskHour;
  String taskDay;

  TaskDTO({
    required this.title,
    required this.description,
    required this.taskHour,
    required this.taskDay,
  });

 

  Map<String, dynamic> toJson() => {
    'title': title,
    'description': description,
    'taskHour': taskHour,
    'taskDay': taskDay
  };
}

File: kotlin/src/main/kotlin/com/taskly/taskly/task/Task.kt
package com.taskly.taskly.task

import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Table
import java.time.LocalDate
import java.time.LocalTime

@Table("TASKS")
data class Task(@Id val id: String? = null,
                var title: String,
                var description: String,
                var taskHour: LocalTime,
                var taskDay: LocalDate,
                var userId: String
)

data class TaskEntity(
    val title: String,
    val description: String,
    val taskHour: LocalTime,
    val taskDay: LocalDate
)

